File: VGTU-Aventory-Backend/src/main/java/org/example/vgtuaventory/model/Sale.java
package org.example.vgtuaventory.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Sale {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @ManyToOne
    private User owner;
    private String saleLocation;
    private LocalDateTime saleDate;
    private Double saleProductPrice;
    private String saleNote;
    private int quantity;
    @ManyToOne
    private Product product;
    private Double totalPrice;
    @Enumerated(EnumType.STRING)
    private SaleType saleType;
}

File: VGTU-Aventory-Frontend/src/features/sales/types/sale.ts
export type SaleType = "SALE" | "RETURN";

export interface SaleRequest {
    productId: number;
    ownerId: number;
    quantity: number;
    saleLocation: string;
    saleType: SaleType;
    saleNote: string;
}

export interface Sale extends SaleRequest {
    id: number;
    saleProductPrice: number;
    totalPrice: number;
    createdAt?: string;
}

export interface User {
    id: number;
    email: string;
    password?: string;
    role: string;
}

export interface ProductWithOwner {
    productId: number;
    productName: string;
    owner: User | null;
    price: number;
    quantity: number;
}
